test/unit/setting_test.rb

```diff
@@ -235,7 +235,8 @@ def test_boolean_true_returns_from_cache
   end
 
   test "foreman_url must be a URI" do
-    assert Setting.find_or_create_by_name(:name => "foreman_url", :default => "http://foo.com")
+    attrs = { :name => "foreman_url", :default => "http://foo.com" }
```

Comment:
`.where(attrs).first_or_create` won't work here because we override `Setting.create`